packages/paste-website/src/components/Heading.tsx

```diff
@@ -94,14 +96,40 @@ const StyledAnchorHyperlink = styled.a`
   }
 `;
 
+const StyledHeaderTag = styled.span`
+  margin-left: ${themeGet('space.space40')};
+  padding: ${themeGet('space.space10')} ${themeGet('space.space20')};
+  font-size: ${themeGet('fontSizes.fontSize20')};
+  font-weight: ${themeGet('fontWeights.fontWeightNormal')};
+  text-transform: uppercase;
+  background-color: ${themeGet('colors.colorGray30')};
+  border-radius: ${themeGet('radii.borderRadius10')};
+`;
+
 const AnchoredHeading: React.FC<HeadingProps> = props => {
   // Only generate slugs for headings where children is 'string'
   if (typeof props.children === 'string') {
     const id = slugify(props.children);
+
+    // get tag prop and add flexStyles if tag prop
+    let flexStyles;
+    let {tag} = props;
+    if (tag) {
```

Comment:
I'm a little uneasy about this implementation. I'd really like to see it as compositions

```
<AnchoredHeading>
  Some text for a heading <Badge>Release 0.3</Badge>
</AnchoredHeading>
```

I get it that isn't a little tricker for mdx, but if we go down this route it might blow up pretty quickly as we add more types of things we might want in the heading

Reply:
@SiTaggart I feel like we should pull this code out, and just not do the tag for now. That way we can do an official kickoff for a badge component.